substrate/frame/nomination-pools/src/weights.rs

```diff
@@ -1249,9 +1330,56 @@ impl WeightInfo for () {
 		// Proof Size summary in bytes:
 		//  Measured:  `901`
 		//  Estimated: `4764`
-		// Minimum execution time: 73_829_000 picoseconds.
-		Weight::from_parts(75_966_000, 4764)
+		// Minimum execution time: 70_284_000 picoseconds.
+		Weight::from_parts(71_375_000, 4764)
 			.saturating_add(RocksDbWeight::get().reads(4_u64))
 			.saturating_add(RocksDbWeight::get().writes(2_u64))
 	}
+	/// Storage: `NominationPools::PoolMembers` (r:1 w:0)
+	/// Proof: `NominationPools::PoolMembers` (`max_values`: None, `max_size`: Some(237), added: 2712, mode: `MaxEncodedLen`)
+	fn apply_slash() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `694`
+		//  Estimated: `3702`
+		// Minimum execution time: 13_403_000 picoseconds.
+		Weight::from_parts(14_064_000, 3702)
+			.saturating_add(RocksDbWeight::get().reads(1_u64))
+	}
+	/// Storage: `NominationPools::PoolMembers` (r:1 w:0)
+	/// Proof: `NominationPools::PoolMembers` (`max_values`: None, `max_size`: Some(237), added: 2712, mode: `MaxEncodedLen`)
+	fn apply_slash_fail() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `732`
+		//  Estimated: `3702`
+		// Minimum execution time: 14_419_000 picoseconds.
+		Weight::from_parts(15_004_000, 3702)
+			.saturating_add(RocksDbWeight::get().reads(1_u64))
+	}
+	fn pool_migrate() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `0`
+		//  Estimated: `0`
+		// Minimum execution time: 759_000 picoseconds.
+		Weight::from_parts(819_000, 0)
+	}
```

Comment:
how does a pool migrate read and write nothing?

also we run the benchmark on master and `apply_slash` `apply_slash_fail`, `pool_migrate` and `migrate_delegation` all read and write nothing. when run against kitchen sink. I think something is wrong no?

https://github.com/paritytech/polkadot-sdk/pull/6418/commits/4dfe581539971010fb462ae1987cdb0a2a2ff734#diff-de1639ae973d95b58521e7385de22a790c15c6a2000b2034bd8f10f5e92b33c5

Reply:
Pool has two strategies, the initial one `TransferStake` and the new `DelegateStake`. Unless it is set to `DelegateStake`, we don't allow migrating (there is nothing to migrate), or slashing (In TS slashing is greedy, ie whole pool is slashed instantly). Since prod is still TS, I left kitchensink with the same setup.

The new setup was tested in Westend and there the weight should have writes.

Reply:
ok thank you